packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -358,17 +365,100 @@ export class GutenbergEditorPage {
 	async closeSettings(): Promise< void > {
 		// On mobile, the settings panel close button is located on the settings panel itself.
 		if ( envVariables.VIEWPORT_NAME === 'mobile' ) {
-			// {@TODO} Temporary measure, to be unified into framelocator based class
-			// once EditorSettingsSidebarComponent is refactored.
-			const editorSettingsSidebarComponent = new EditorSettingsSidebarComponent(
-				await this.getEditorFrame(),
-				this.page
-			);
-			await editorSettingsSidebarComponent.closeSidebar();
+			await this.editorSettingsSidebarComponent.closeSidebarForMobile();
+		} else {
+			await this.editorToolbarComponent.closeSettings();
 		}
-		await this.editorToolbarComponent.closeSettings();
 	}
 
+	/**
+	 * Clicks and activates the given tab in the Editor Settings sidebar.
+	 *
+	 * @param {EditorSidebarTab} tab Name of the tab to activate.
+	 */
+	async clickSettingsTab( tab: EditorSidebarTab ): Promise< void > {
+		await this.editorSettingsSidebarComponent.clickTab( tab );
+	}
+
+	/**
+	 * Sets the article's privacy/visibility option.
+	 *
+	 * @param {PrivacyOptions} visibility Visibility option for the article.
+	 * @param param1 Object parameters.
+	 * @param {string} param1.password Password for the post.
+	 */
+	async setArticleVisibility(
+		visibility: PrivacyOptions,
+		{ password }: { password?: string }
+	): Promise< void > {
+		await Promise.race( [
+			this.editorSettingsSidebarComponent.clickTab( 'Page' ),
+			this.editorSettingsSidebarComponent.clickTab( 'Post' ),
+		] );
```

Comment:
You'll see this construct used throughout this PR. It's meant to make the methods that perform a series of actions to be page/post agnostic. 

The alternative was to make the calling function (which would be the test step in this case) explicitly activate the tab prior to executing actions, but I feel it doesn't make sense to do so from both a usability and logical sense.